sdk/tables/azure-data-tables/inc/azure/data/tables/table_client.hpp

```diff
@@ -236,19 +186,21 @@ namespace Azure { namespace Data { namespace Tables {
      * @return Table entity.
      */
     Response<Models::TableEntity> GetEntity(
-        const std::string& partitionKey,
-        const std::string& rowKey,
+        std::string const& partitionKey,
+        std::string const& rowKey,
         Core::Context const& context = {});
 
     /**
      * @brief Submits a transaction.
      *
-     * @param steps the transaction steps to execute.
+     * @param steps The transaction steps to execute.
+     * @param batchId The batchId for the transaction.
      * @param context for canceling long running operations.
      * @return Submit transaction result.
      */
     Response<Models::SubmitTransactionResult> SubmitTransaction(
         std::vector<Models::TransactionStep> const& steps,
+        std::string const& batchId = Azure::Core::Uuid::CreateUuid().ToString(),
         Core::Context const& context = {});
```

Comment:
I don't see a concept of batchId exposed publicy in the .NET/Java SDKs. What motivated us adding this?
https://github.com/Azure/azure-sdk-for-net/blob/8a63175dbbebe20415c8dc175fa9ae64296b3c1e/sdk/tables/Azure.Data.Tables/api/Azure.Data.Tables.netstandard2.0.cs#L53
https://github.com/Azure/azure-sdk-for-java/blob/031d19ffec27c3dda5ffc650eb3a5badcfc4cbdf/sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableClient.java#L1481-L1484

I do see the multi part boundary being set to a uuid, for multipart here, so maybe this should just be an implementation detail?
https://github.com/Azure/azure-sdk-for-net/blob/8a63175dbbebe20415c8dc175fa9ae64296b3c1e/sdk/storage/Azure.Storage.Blobs.Batch/src/Multipart.cs#L63

Reply:
because test proxy , if the batchid is randomly generated , then i cannot playback recordings as the things change from one batch to the other, that is why i made it have a default , but i can also pass it from the tests  
Unless, just thinking here, i make the implementation a protected, the public does what it did before , the protected takes the batch id param, and the tests are friends. But that seems a bit to overengineered for passing in a string, that customers could also use for debuggin / tracing purposes

Reply:
Generally, we shouldn't introduce complexity into customer-facing public API surface changes, only for our testing scenarios.
We've avoided doing so thus far, and none of the other languages have done that (who presumably have the same testing concern).

Some of the alternatives you were exploring (around private helper method/make tests a friend) which don't impact the public surface area are reasonable to pursue, as needed. We have precedence for that pattern elsewhere in the SDK.